### PR TITLE
feat: xcall upgradeable

### PR DESCRIPTION
**Pull request by gcranju**
### Commit Message

```bash
feat: xcall upgradeable
```

see the [guidelines](https://github.com/icon-project/community/blob/main/guidelines/technical/software-development-guidelines.md#commit-messages) for commit messages.

### Changelog Entry

```bash
version: <log entry>
```

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have documented my code in accordance with the [documentation guidelines](https://github.com/icon-project/community/blob/main/guidelines/technical/software-development-guidelines.md#documentation)
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have run the unit tests
- [ ] I only have one commit (if not, squash them into one commit).
- [ ] I have a descriptive commit message that adheres to the [commit message guidelines](https://github.com/icon-project/community/blob/main/guidelines/technical/software-development-guidelines.md#commit-messages)

> Please review the [CONTRIBUTING.md](/CONTRIBUTING.md) file for detailed contributing guidelines.

### CODE REVIEW

**Comment by izyak** [comment]
I don't think this is a safe way to upgrade contracts.
Can you reference this please?

https://github.com/OpenZeppelin/openzeppelin-foundry-upgrades

**Review by AntonAndell — CHANGES_REQUESTED** [review]

File: contracts/evm/contracts/xcall/CallService.sol [hunk 1]
```diff
@@ -523,4 +525,10 @@ contract CallService is IBSH, ICallService, IFeeManage, Initializable {
     function verifySuccess(uint256 _sn) external view returns (bool) {
         return successfulResponses[_sn];
     }
+    
+        /* ========== UUPS ========== */
+    //solhint-disable-next-line no-empty-blocks
+    function _authorizeUpgrade(address) internal override onlyAdmin {}
+
+    function getImplementation() external view returns (address) {}
```
Comment (AntonAndell) [inline contracts/evm/contracts/xcall/CallService.sol:533]: What does this function do?
Should this do something like:
return ERC1967Utils.getImplementation();?

File: contracts/evm/contracts/upgradeable/UUPSProxy.sol [hunk 2]
```diff
@@ -0,0 +1,10 @@
+// SPDX-License-Identifier: MIT
```
Comment (AntonAndell) [inline contracts/evm/contracts/upgradeable/UUPSProxy.sol:1]: Can we use ERC1967Proxy directly? instead of having this empty implementation?

Comment (gcranju) [reply contracts/evm/contracts/upgradeable/UUPSProxy.sol]: We changed upgradaebility to Transparent upgrade so, this is removed!

File: contracts/evm/contracts/xcall/CallService.sol [hunk 3]
```diff
@@ -523,4 +525,10 @@ contract CallService is IBSH, ICallService, IFeeManage, Initializable {
     function verifySuccess(uint256 _sn) external view returns (bool) {
         return successfulResponses[_sn];
     }
+    
+        /* ========== UUPS ========== */
+    //solhint-disable-next-line no-empty-blocks
+    function _authorizeUpgrade(address) internal override onlyAdmin {}
```
[the remark this answers is not in the document]
Comment (gcranju) [reply contracts/evm/contracts/xcall/CallService.sol]: Now this will be decided when deploying contract!

**Comment by izyak** [comment]
Looks good to me! Works as expected. Okay to merge.

**Review by izyak — APPROVED** [review]

**Review by AntonAndell — APPROVED** [review]
Stucture looks good. And Newton tested deployment and upgrades. Should be good to go

**Pull request merged**